v2/git-initializer/main.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/cache"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	gitssh "github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/go-git/go-git/v5/storage/filesystem"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/brigadecore/brigade/sdk/v2/core"
+)
+
+func main() {
+	if err := gitCheckout(); err != nil {
+		fmt.Printf("\n%s\n\n", err)
+		os.Exit(1)
+	}
+}
+
+// TODO: needs retry - let's do a follow-up (krancour has a lib to use!)
+
+func gitCheckout() error {
+	eventPath := "/var/event/event.json"
+	data, err := ioutil.ReadFile(eventPath)
+	if err != nil {
+		return errors.Wrapf(err, "unable read the event file %q", eventPath)
+	}
+
+	var event struct {
+		Project struct {
+			Secrets map[string]string `json:"secrets"`
+		} `json:"project"`
+		Worker struct {
+			Git *core.GitConfig `json:"git"`
+		} `json:"worker"`
+	}
+	err = json.Unmarshal(data, &event)
+	if err != nil {
+		return errors.Wrap(err, "error unmarshaling the event")
+	}
+
+	// Extract git config
+	gitConfig := event.Worker.Git
+	if gitConfig == nil {
+		return fmt.Errorf("git config from %q is empty", eventPath)
+	}
+
+	// Setup Auth
+	var auth transport.AuthMethod
+	// TODO: What about token-based auth?  (see v1 askpass.sh/BRIGADE_REPO_AUTH_TOKEN)
+
+	// TODO: Check for SSH Cert
+	// https://github.blog/2019-08-14-ssh-certificate-authentication-for-github-enterprise-cloud/
+	//
+	// (BRIGADE_REPO_SSH_CERT in v1)
+	// I think all we need to do is to make sure the cert file exists at /id_dsa-cert.pub
```

Comment:
Remind me... when is this needed? It's just when the root CA for the git server isn't already trusted?

Reply:
It's an optional feature and support for it exists in the v1 git-sidecar, thus I figured for parity we'd want to continue support.

Ref https://github.com/brigadecore/brigade/pull/1008

Reply:
I know v1 accommodated this. I'm just experiencing some kind of mental block in understanding it.

Reply:
This is non-blocking... it was for my own edification.